### ISSUE TITLE
Problem with negative integer argument

### ISSUE THREAD

**Issue by Sciss** [scallop/scallop#189]
I was happy to see that scallop parses negative doubles and integers; but it seems to break for certain cases. For example:

```scala
import org.rogach.scallop._

object c extends ScallopConf(Seq("-1234")) {
  val value = trailArg[Int]()
  val foo = opt[Boolean]()
}

c.verify()
```
```
[scallop] Error: Unknown option '-1234'
```

Whereas if I use the flag, it works:

```scala
object c extends ScallopConf(Seq("--foo", "-1234")) {
  val value = trailArg[Int]()
  val foo = opt[Boolean]()
}

c.verify(); c.value()
```
```
res1: Int = -1234
```

**Comment by Rogach** [comment]
Scallop supports "number-only" options (similar to linux `head` and `tail`, which take numbers with dashes as options, for example `head -10`). Thus Scallop didn't support `-N` values in trailing args.

However, looking at the code now I think I can add support for such values without too much problems.

**Comment by Rogach** [comment]
Released version `3.3.0` with the fix.

**Comment by Sciss** [comment]
thanks!
